GTBitsOfGood/canine-assistants
# Upload and Delete Images Backend

## Image Uploads and Deleting

Issue Number(s): #107

Created backend API endpoint and actions for uploading and deleting images in Backblaze and connected them to the frontend ImageUpload component. Upload an image by either creating a new dog or editing an existing one. The upload API endpoint also sets the dog's image URL as the uploaded image. An image can be deleted by editing an existing dog and hitting `delete` then `save`. Images are also deleted when a dog is deleted. The seed file has been updated as well so that each dog existing dog has the default field `image: ""`, so make sure to seed the database before testing.

### Checklist


### Related PRs

PR Number(s): #108 

### How To Test

Log into [Backblaze.com](https://www.backblaze.com/) with information from Bitwarden, find using `bw list items --search backblaze`, see password corresponding to gtbitsofgood.**dev**@gmail.com (NOT gtbitsofgood.**aau**@gmail.com)

Add these values to `.env` file:
`BACKBLAZE_BUCKET_ID=`[found on Backblaze website under Bucket ID on first page]
`BACKBLAZE_KEY_ID=`[found on Backblaze website on Account > Application Keys > Your Application Keys > keyID, note that this is NOT the master key]
`BACKBLAZE_APP_KEY=`[found in Bitwarden under appKey]

Make sure that information is correct by calling `/api/images` to see if `success` is true or an error message. Don't hesitate to reach out if there are any problems!
